pkg/services/accesscontrol/database/resource_permissions.go

```diff
@@ -32,7 +32,11 @@ func (p *flatResourcePermission) Managed() bool {
 	return strings.HasPrefix(p.RoleName, "managed:")
 }
 
-func (s *AccessControlStore) SetUserResourcePermission(ctx context.Context, orgID, userID int64, cmd accesscontrol.SetResourcePermissionCommand) (*accesscontrol.ResourcePermission, error) {
+func (s *AccessControlStore) SetUserResourcePermission(
+	ctx context.Context, orgID, userID int64,
+	cmd accesscontrol.SetResourcePermissionCommand,
+	hook func(session *sqlstore.DBSession, orgID, userID int64, resourceID, permission string) error,
```

Comment:
nit: could we get a type alias for the hook func to improve readability and cross reference?